tests/shared/drive-constraints.spec.js

```diff
@@ -303,52 +303,219 @@ describe('Shared: DriveConstraints', function() {
+            path: path.join(__dirname, 'rpi.img'),
+            size: {
+              original: this.drive.size,
+              final: {
+                estimation: false
+              }
+            }
+          };
+        });
+
+        it('should return true if the final size is less than the drive size', function() {
+          this.image.size.final.value = this.drive.size - 1;
+          m.chai.expect(constraints.isDriveLargeEnough(this.drive, this.image)).to.be.true;
+        });
+
+        it('should return true if the final size is equal to the drive', function() {
+          this.image.size.final.value = this.drive.size;
+          m.chai.expect(constraints.isDriveLargeEnough(this.drive, this.image)).to.be.true;
+        });
+
+        it('should return false if the final size is greater than the drive size', function() {
+          this.image.size.final.value = this.drive.size + 1;
+          m.chai.expect(constraints.isDriveLargeEnough(this.drive, this.image)).to.be.false;
+        });
+
+      });
+
+      describe('given the original size is greater than the drive size', function() {
+
+        beforeEach(function() {
+          this.image = {
+            path: path.join(__dirname, 'rpi.img'),
+            size: {
+              original: this.drive.size + 1,
+              final: {
+                estimation: false
+              }
+            }
+          };
+        });
+
+        it('should return true if the final size is less than the drive size', function() {
+          this.image.size.final.value = this.drive.size - 1;
+          m.chai.expect(constraints.isDriveLargeEnough(this.drive, this.image)).to.be.true;
+        });
+
+        it('should return true if the final size is equal to the drive', function() {
+          this.image.size.final.value = this.drive.size;
+          m.chai.expect(constraints.isDriveLargeEnough(this.drive, this.image)).to.be.true;
+        });
+
+        it('should return false if the final size is greater than the drive size', function() {
+          this.image.size.final.value = this.drive.size + 1;
+          m.chai.expect(constraints.isDriveLargeEnough(this.drive, this.image)).to.be.false;
+        });
 
-    it('should return true if the drive size is equal to the image size', function() {
-      const result = constraints.isDriveLargeEnough({
-        device: '/dev/disk1',
-        name: 'USB Drive',
-        size: 1000000000,
-        protected: false
-      }, {
-        path: 'rpi.img',
-        size: 1000000000
       });
 
-      m.chai.expect(result).to.be.true;
     });
 
-    it('should return false if the drive size is less than the image size', function() {
-      const result = constraints.isDriveLargeEnough({
-        device: '/dev/disk1',
-        name: 'USB Drive',
-        size: 1000000000,
-        protected: false
-      }, {
-        path: 'rpi.img',
-        size: 1000000001
+    describe('given the final image size estimation flag is true', function() {
```

Comment:
Would 'given the final image size is an estimation' be more readable? Or is it better to have the test descriptions be brutally accurate?